Matrix tests

siddC opened the pull request:
* [x] fix #233 
* [x] description of feature/fix
* [x] tests added/passed
* [x] add an entry to the [next release](../HISTORY.rst)

### Summary

These changes should address the concerns in issue #233 except for adding in/testing for a list of parallel routes. Additionally, the number of independent conserved metabolite relations, i.e. conservation pools, is added but the actual pools themselves are not shown as the left-nullspace of large network reconstructions is extremely difficult to interpret.

Midnighter commented on memote/support/matrix.py:
```diff
@@ -0,0 +1,93 @@
+    s_matrix, _, _ = con_helpers.stoichiometry_matrix(
+        model.metabolites, model.reactions
+    )
+    abs_matrix, _, _ = abs(s_matrix)
```
`s_matrix` is a 2D `ndarray` so I don't understand why its absolute matrix should return a triple.

Midnighter commented on memote/support/matrix.py:
```diff
@@ -0,0 +1,93 @@
+    abs_matrix, _, _ = abs(s_matrix)
+
+    absolute_max_coef = abs_matrix.max()
+    absolute_non_zero_min_coef = np.min(abs_matrix[np.nonzero(abs_matrix)])
```
Since you're dealing with the absolute values `abs_matrix[abs_matrix > 0].min()` should be fine.

Midnighter commented on memote/support/matrix.py:
```diff
@@ -0,0 +1,93 @@
+    free variables in the system. The forumla used calculates this using the
+    rank-nullity theorem. For more information, see the links below.
+
+    See Also:
```
`See Also` is reserved for pointers to other related functions in the package. Please use a `References` section here (cf. https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#sections point 11).

siddC commented:
I've changed the logic in `absolute_extreme_coefficient_ratio` with regards to min, max, and absolute value. I've also fixed the docstring for `degrees_of_freedom` and added in the test changes into `test_config.yml` and in `HISTORY.rst`. Let me know if anymore changes need to be made to this branch.

Midnighter commented:
Nice work @siddC! I like your commit structuring and their messages.
